Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,342 @@
+
+# Description
+
+Qubits are a managed resource in Q#: when qubits are allocated, they are valid only within the new block declared at allocation, after which they are released and can no longer be used.
+This proposal makes qubit allocation in Q# always scoped to the current block.
+Instead of being released at the end of a newly-declared block, qubits are released at the end of the current block.
+It also extends block statement syntax to allow precise control of qubit lifetime when needed in more complicated scenarios.
+Finally, this proposal aims to be syntactically minimal and consistent with other binding statements like `let`, in contrast to alternatives that have more complicated syntax.
+
+## Current Status
+
+Currently, the only mechanism to allocate qubits in Q# is with `using` and `borrowing` block statements.
+These statements will be deprecated and removed, replaced with new `use` and `borrow` statements that do not require a block, as well as a new scope statement that can be used when a new block is desired.
+
+No new functionality is provided by this proposal.
+Every example with the existing syntax has a corresponding equivalent example with the proposed syntax, and vice versa.
+However, the proposed syntax addresses several problems with the existing syntax:
+
+1. Example 1a: The lifetime of the qubit `q` is the same as the lifetime of the surrounding block.
+   A new block is syntactically required, but does not serve any purpose other than increasing the indentation level.
+2. Example 2a: Allocating multiple qubits requires either a single `using` statement with tuple destructuring, or multiple nested `using` statements.
+   Tuple destructuring is hard to read when many variables are declared or when the qubit initializer expression is long.
+   It is hard to tell which tuple item corresponds to which variable name.
+3. Example 3a: Allocating multiple qubits with nested `using` statements makes it clear which qubit initializer expression corresponds to which variable name, but at the cost of many levels of unnecessary indentation.
+4. Example 4a: When `using` and `borrowing` statements are mixed, nested blocks are required; tuple destructuring is not possible.
+
+Example 5a shows the main benefit of the current syntax: it allows the lifetime of a qubit to be shorter than the parent block.
+While this exact syntax will be removed, the proposed syntax can also express this in a simple way.
+
+### Examples
+
+Example 1a: A single qubit allocated and released at the same time as the end of the outer block.
+
+```qsharp
+operation FlipCoin() : Result {
+    using (q = Qubit()) {
+        H(q);
+        return M(q);
+    }
+}
+```
+
+Example 2a: Many qubits allocated in a single `using` statement with tuple destructuring.
+
+```qsharp
+operation QubitTuple(n : Int) : Result {
+    using ((a, b, c) = (Qubit(), Qubit[2 * n + 1], Qubit[n])) {
+        // ...
+        return M(q);
```

Comment:
Since `q` is not one of the qubits allocated in the example, it is a bit confusing to include it here. It is possible the qubit was allocated somewhere else, but it may be simpler to replace this reference to refer to one of the qubits that are part of the example, like `a`.